### DISCUSSION TITLE
How can I customize url?

### DISCUSSION THREAD

**Discussion by morganealemany** [rappasoft/laravel-livewire-tables#1548] [Q&A] [answered]
I used to work with laravel-livewire-tables version ^2.0 (with livewire 2.0) and I upgrade to ^3.0 version (and livewire to 3.0 version). In livewire 3.0 the asset_url config doesn't exist anymore and I have to add into routes/web.php :

`    Livewire::setScriptRoute(function ($handle) {
        return Route::get('/example/public/livewire/livewire.js', $handle);
    });
`
instead of 
` 'asset_url' => 'http://xxx.xxx.x.xxx/example/public',` in config/livewire.php

It seems that livewire-tables doesn't use the Livewire::setScriptRoute() because I have these errors :
` GET http://xxx.xxx.x.xxx/rappasoft/laravel-livewire-tables/core.min.css net::ERR_ABORTED 404 (Not Found)

       GET http://xxx.xxx.x.xxx/rappasoft/laravel-livewire-tables/thirdparty.css net::ERR_ABORTED 404 (Not Found)

       GET http://xxx.xxx.x.xxx/rappasoft/laravel-livewire-tables/core.min.js net::ERR_ABORTED 404 (Not Found)
   
       GET http://xxx.xxx.x.xxx/rappasoft/laravel-livewire-tables/thirdparty.min.js net::ERR_ABORTED 404 (Not Found)`

Is there a way to customize the url?

**Comment by lrljoe** [comment]
Not in this version, (Not easily anyway).

it's been on my to do list for a bit, so will make it into the next release most likely.

You can however use the following methods, which respect the format of Livewire::setScriptRoute, but your mileage may vary!
Livewire::setRappasoftTableScriptRoute
Livewire::setRappasoftTableStylesRoute
Livewire::setRappasoftTableThirdPartyScriptRoute
Livewire::setRappasoftTableThirdPartyStylesRoute

Can I ask the issue you're experiencing here though, as those paths are used when injection is used, if you're publishing the code and including it yourself in another way, then you can ignore all of that completely (both for Livewire and Tables).

I'm assuming you've just got the whole site on a subdirectory?

**Reply by lrljoe** [reply]
#1549 <- should fix this, just need to write some tests...

**Reply by morganealemany** [reply]
I have the whole site on a subdirectory indeed on my development server. But on production server it's not.
So I have these errors only on development.

**Reply by lrljoe** [reply, marked as answer]
This should now be available in v3.1.3 in the config file, e.g:
```
    /**
     * Customise Script & Styles Paths
     */
    'script_base_path' => '/test_system/rappasoft/laravel-livewire-tables',
```

**Reply by morganealemany** [reply]
Great job!! Thanks a lot ;)
